--- a/beacon_chain/version.nim
+++ b/beacon_chain/version.nim
@@ -15,7 +15,7 @@
 const
   compileYear = CompileDate[0 ..< 4]  # YYYY-MM-DD (UTC)
   copyrights* =
-    "Copyright (c) 2019-" & compileYear & " Status Research & Development GmbH"
+    "Copyright (c) 2019-2024" & " Status Research & Development GmbH"
 
   versionMajor* = 24
   versionMinor* = 3
@@ -41,7 +41,8 @@
   else:
     git_revision_override
 
-  gitRevision* = strip(gitRevisionLong)[0..5]
+  # bake in gitRevision as git is not avaialable at build
+  gitRevision* = "dc19b0"
 
   nimFullBanner* = staticExec("nim --version")
   nimBanner* = staticExec("nim --version | grep Version")
