--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 EF_TESTS = "testing/ef_tests"
 STATE_TRANSITION_VECTORS = "testing/state_transition_vectors"
 EXECUTION_ENGINE_INTEGRATION = "testing/execution_engine_integration"
-GIT_TAG := $(shell git describe --tags --candidates 1)
+GIT_TAG := $(shell git describe --tags --candidates 1 2> /dev/null)
 BIN_DIR = "bin"
 
 X86_64_TAG = "x86_64-unknown-linux-gnu"
@@ -219,6 +219,6 @@
 
 # Performs a `cargo` clean and cleans the `ef_tests` directory.
 clean:
-	cargo clean
-	make -C $(EF_TESTS) clean
-	make -C $(STATE_TRANSITION_VECTORS) clean
+	-cargo clean
+	$(MAKE) -C $(EF_TESTS) clean
+	$(MAKE) -C $(STATE_TRANSITION_VECTORS) clean 
--- a/testing/state_transition_vectors/Makefile
+++ b/testing/state_transition_vectors/Makefile
@@ -5,4 +5,4 @@
 	cargo test --release
 
 clean:
-	rm -r vectors/
+	-rm -rf vectors/
