--- a/hack/workspace_status.sh
+++ b/hack/workspace_status.sh
@@ -1,9 +1,9 @@
 #!/bin/bash
 
 # Note: The STABLE_ prefix will force a relink when the value changes when using rules_go x_defs.
-
-echo STABLE_GIT_COMMIT "$(git rev-parse HEAD)"
-echo DATE "$(date --rfc-3339=seconds --utc)"
-echo DATE_UNIX "$(date --utc +%s)"
+# Patch for reproducibility
+echo STABLE_GIT_COMMIT "bc9151d647e70cf5e0b6dbcee01998c430a96cb5"
+echo DATE "2024-04-30 01:44:55+00:00"
+echo DATE_UNIX "1714441523"
 echo DOCKER_TAG "$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short=6 HEAD)"
 echo STABLE_GIT_TAG "$(git describe --tags --abbrev=0)"
