Index: eth-node-lodestar-1.19.0/packages/cli/src/cli.ts
===================================================================
--- eth-node-lodestar-1.19.0.orig/packages/cli/src/cli.ts
+++ eth-node-lodestar-1.19.0/packages/cli/src/cli.ts
@@ -9,7 +9,7 @@ import {getVersionData} from "./util/ver
 const {version} = getVersionData();
 const topBanner = `🌟 Lodestar: TypeScript Implementation of the Ethereum Consensus Beacon Chain.
   * Version: ${version}
-  * by ChainSafe Systems, 2018-${new Date().getFullYear()}`;
+  * by ChainSafe Systems, 2018-2024`;
 const bottomBanner = `📖 For more information, check the CLI reference:
   * https://chainsafe.github.io/lodestar/reference/cli
 
Index: eth-node-lodestar-1.19.0/packages/cli/src/util/version.ts
===================================================================
--- eth-node-lodestar-1.19.0.orig/packages/cli/src/util/version.ts
+++ eth-node-lodestar-1.19.0/packages/cli/src/util/version.ts
@@ -47,6 +47,13 @@ export function getVersionData(): {
     if (!localVersion || !localVersion.includes(commitShort)) {
       parts.push(commitShort);
     }
+  }else {
+    const commitShort = process.env.GIT_COMMIT;
+    if (commitShort){
+       parts.push(commitShort)
+    } else {
+       parts.push("c5b4c1c")
+    }
   }
 
   return {
Index: eth-node-lodestar-1.19.0/packages/flare/src/cli.ts
===================================================================
--- eth-node-lodestar-1.19.0.orig/packages/flare/src/cli.ts
+++ eth-node-lodestar-1.19.0/packages/flare/src/cli.ts
@@ -10,7 +10,7 @@ Flare is a sudden brief burst of bright
 In the wrong hands, can lead people astray.
 Use with care.
 
-  * by ChainSafe Systems, 2018-${new Date().getFullYear()}`;
+  * by ChainSafe Systems, 2018-2024`;
 const bottomBanner = `
 ✍️ Give feedback and report issues on GitHub:
   * https://github.com/ChainSafe/lodestar`;
Index: eth-node-lodestar-1.19.0/packages/prover/src/cli/cli.ts
===================================================================
--- eth-node-lodestar-1.19.0.orig/packages/prover/src/cli/cli.ts
+++ eth-node-lodestar-1.19.0/packages/prover/src/cli/cli.ts
@@ -9,7 +9,7 @@ import {globalOptions} from "./options.j
 const {version} = getVersionData();
 const topBanner = `🌟 Lodestar Prover Proxy: Ethereum RPC proxy for RPC responses, verified against the trusted block hashes.
   * Version: ${version}
-  * by ChainSafe Systems, 2018-${new Date().getFullYear()}`;
+  * by ChainSafe Systems, 2018-2024`;
 const bottomBanner = `📖 For more information, check the CLI reference:
   * https://chainsafe.github.io/lodestar/reference/cli
 
Index: eth-node-lodestar-1.19.0/packages/prover/src/utils/version.ts
===================================================================
--- eth-node-lodestar-1.19.0.orig/packages/prover/src/utils/version.ts
+++ eth-node-lodestar-1.19.0/packages/prover/src/utils/version.ts
@@ -37,6 +37,7 @@ export function getVersionData(): {
 
   const {branch, commit} = readAndGetGitData();
 
+
   // Add branch only if not present and not an ignore value
   if (branch && !BRANCH_IGNORE.test(branch)) parts.push(branch);
 
@@ -47,6 +48,13 @@ export function getVersionData(): {
     if (!localVersion || !localVersion.includes(commitShort)) {
       parts.push(commitShort);
     }
+  } else {
+    const commitShort = process.env.GIT_COMMIT;
+    if (commitShort){
+       parts.push(commitShort)
+    } else {
+       parts.push("c5b4c1c")
+    }
   }
 
   return {
