Index: erigon/fix-service-config.sh
===================================================================
--- erigon.orig/fix-service-config.sh	2023-11-13 10:38:37.421385488 +0000
+++ erigon/fix-service-config.sh	2023-11-13 10:41:01.954518524 +0000
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # List of keywords to be quoted
-keywords=("private.api.addr" "http.api" "http.port" "authrpc.port" "torrent.port")
+keywords=("private.api.addr" "http.api" "http.port" "authrpc.port" "torrent.port" "authrpc.jwtsecret")
 
 # Your TOML file
 toml_file="/etc/eth-node-erigon-service-sepolia/config.toml"
