--- a/vendor/nimbus-build-system/scripts/build_nim.sh
+++ b/vendor/nimbus-build-system/scripts/build_nim.sh
@@ -77,9 +77,9 @@
 		# We can't use "rev-parse" here, because it would return the tag object's
 		# hash instead of the commit hash, when NIM_COMMIT is a tag.
 		NIM_COMMIT_HASH="$(git rev-list -n 1 ${NIM_COMMIT})"
-	else
+	#else
 		# NIM_COMMIT is empty, so assume the commit we need is already checked out
-		NIM_COMMIT_HASH="$(git rev-list -n 1 HEAD)"
+		#NIM_COMMIT_HASH="$(git rev-list -n 1 HEAD)"
 	fi
 	popd >/dev/null
 
--- a/vendor/nimbus-build-system/makefiles/targets.mk
+++ b/vendor/nimbus-build-system/makefiles/targets.mk
@@ -78,7 +78,7 @@
 #- macOS is also a special case, with its "ln" not supporting "-r"
 #- the AppVeyor 32-bit build is done on a 64-bit image, so we need to override the architecture detection with ARCH_OVERRIDE
 build-nim: | sanity-checks
-	+ if [[ -z "$(NIM_COMMIT)" ]]; then git submodule update --init --recursive "$(BUILD_SYSTEM_DIR)"; fi; \
+	+ #if [[ -z "$(NIM_COMMIT)" ]]; then git submodule update --init --recursive "$(BUILD_SYSTEM_DIR)"; fi;
 		NIM_BUILD_MSG="$(BUILD_MSG) Nim compiler" \
 		V=$(V) \
 		CC=$(CC) \
@@ -125,7 +125,7 @@
 
 # don't use this target, or you risk updating dependency repos that are not ready to be used in Nimbus
 update-remote:
-	git submodule update --remote
+	#git submodule update --remote
 
 nat-libs: | libminiupnpc.a libnatpmp.a
 
@@ -149,8 +149,8 @@
 #  for runtime path (i.e.: the second line in $(NIMBLE_DIR)/pkgs/*/*.nimble-link)
 $(NIMBLE_DIR):
 	mkdir -p $(NIMBLE_DIR)/pkgs
-	NIMBLE_DIR="$(CURDIR)/$(NIMBLE_DIR)" PWD_CMD="$(PWD)" EXCLUDED_NIM_PACKAGES="$(EXCLUDED_NIM_PACKAGES)" \
-		git submodule foreach --recursive --quiet '$(CURDIR)/$(BUILD_SYSTEM_DIR)/scripts/create_nimble_link.sh "$$sm_path"'
+	NIMBLE_DIR="$(CURDIR)/$(NIMBLE_DIR)" PWD_CMD="$(PWD)" EXCLUDED_NIM_PACKAGES="$(EXCLUDED_NIM_PACKAGES)"
+		#git submodule foreach --recursive --quiet '$(CURDIR)/$(BUILD_SYSTEM_DIR)/scripts/create_nimble_link.sh "$$sm_path"'
 
 clean-cross:
 	+ [[ -e vendor/nim-nat-traversal/vendor/miniupnp/miniupnpc ]] && "$(MAKE)" -C vendor/nim-nat-traversal/vendor/miniupnp/miniupnpc CC=$(CC) clean $(HANDLE_OUTPUT) || true
