--- a/Makefile
+++ b/Makefile
@@ -3,9 +3,9 @@
 UNAME = $(shell uname) # Supported: Darwin, Linux
 DOCKER := $(shell command -v docker 2> /dev/null)
 
-GIT_COMMIT ?= $(shell git rev-list -1 HEAD)
-GIT_BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
-GIT_TAG    ?= $(shell git describe --tags '--match=v*' --dirty)
+GIT_COMMIT ?= $(shell git rev-list -1 HEAD 2> /dev/null)
+GIT_BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD 2> /dev/null)
+GIT_TAG    ?= $(shell git describe --tags '--match=v*' --dirty 2> /dev/null)
 ERIGON_USER ?= erigon
 # if using volume-mounting data dir, then must exist on host OS
 DOCKER_UID ?= $(shell id -u)
