Index: eth-node-reth-1.1.1/Makefile
===================================================================
--- eth-node-reth-1.1.1.orig/Makefile
+++ eth-node-reth-1.1.1/Makefile
@@ -62,7 +62,7 @@ install-op: ## Build and install the op-
 build: ## Build the reth binary into `target` directory.
 	cargo build --bin reth --features "$(FEATURES)" --profile "$(PROFILE)"
 
-SOURCE_DATE_EPOCH := $(shell git log -1 --pretty=%ct)
+SOURCE_DATE_EPOCH ?= $(shell git log -1 --pretty=%ct)
 .PHONY: reproducible
 reproducible: ## Build the reth binary into `target` directory with reproducible builds. Only works for x86_64-unknown-linux-gnu currently
 	SOURCE_DATE_EPOCH=$(SOURCE_DATE_EPOCH) \
